python/segyio/trace.py

```diff
@@ -842,9 +843,26 @@ def __setitem__(self, i, val):
 
         >>> f.text[1] = g.text[0]
 
+        If i is slice [i_1:i_n] and val=[val_1, .. val_n],
```

Comment:
This documentation is very confusing to me.

In `__setitem__` for trace headers, the following is written:

> If i or o is a slice, val must be an iterable.

This _very_ briefly describes the behaviour. This is then made obvious by the actual examples.